src/Common/tests/System/Xml/XPath/MiscellaneousCases/MiscellaneousCasesTests.cs

```diff
@@ -373,5 +373,69 @@ public static void MiscellaneousCasesTest5319()
 
             Utils.XPathNodesetTest(xml, testExpression, expected, startingNodePath: startingNodePath);
         }
+
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        public static void WhenUsingCustomXPathNavigatorDescendantIteratorShouldBeDoneAfterIterationIsComplete()
+        {
+            // When XPathDescendantIterator.MoveNext returns false it should keep returning false
+            // This is a regression test.
+            // Previously this scenario was possible to hit only when inheriting from XPathNavigator
+            // and not providing custom XPathDescendantsNavigator
+            string xml = "<root><a/><b/><c/><d><e/><f/></d></root>";
```

Comment:
Isn't it better test siblings after an element having children. (i.e. add a sibling after 'd' element for this test)

Reply:
I think this won't make any difference for the original repro since internal navigator will be on the root after we are finished anyway